extensions/scarb-snforge-test-collector/src/metadata.rs

```diff
@@ -150,11 +167,29 @@ impl CompilationUnit<'_> {
             .unwrap_or(true)
     }
 
-    pub fn source_root(&self) -> Utf8PathBuf {
-        self.unit_metadata.target.source_root().to_path_buf()
+    pub fn main_package_source_root(&self) -> Utf8PathBuf {
+        self.main_package_metadata.source_root().to_path_buf()
     }
 
-    pub fn source_file_path(&self) -> &Utf8Path {
-        &self.unit_metadata.target.source_path
+    pub fn main_package_source_file_path(&self) -> &Utf8Path {
+        &self.main_package_metadata.source_path
+    }
```

Comment:
Changed it bcs we actually want the source root and source file path of the package we are compiling the tests for, not of the target. They are the same for now bcs we take the CU with main package being the package we want to compile tests for, but the logic it is less prone to break in the future this way.
cc @maciektr to check if I'm right

Reply:
Ok, if you don't fully integrate with the Scarb data model it kind of makes sense.